trino/client.py

```diff
@@ -1128,14 +1128,26 @@ def map(self, values: List[Any]) -> Optional[List[Any]]:
         return [self.mapper.map(value) for value in values]
 
 
+class NamedRowTuple(tuple):
+    """Custom tuple class as namedtuple doesn't support missing or duplicate names"""
```

Comment:
IMO the behaviour should be:

- If duplicate fields exist return a plain tuple/throw if named tuple is requested
  - Current behaviour is bad because it leads to making some fields return "wrong" values when accessed by name
- If a field name is missing generate one as `field + str(n)` where `n` is a counter
- If all fields have unique names return a namedtuple

Reply:
I think it should behave like Trino for ambiguous field names. 

```
trino> SELECT CAST(ROW(1, 2.0) AS ROW(x BIGINT, x DOUBLE)).x;
Query 20230306_185114_07832_kzrg9 failed: line 1:53: Ambiguous row field reference: x
```